charms/knative-operator/src/charm.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+import traceback
+
+from ops.main import main
+from ops.pebble import Layer
+from ops.pebble import APIError as pebble_APIError
+from ops.charm import CharmBase
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus
+from jinja2 import Environment, FileSystemLoader
+from lightkube import Client, codecs
+from lightkube import ApiError as lightkube_ApiError
+
+logger = logging.getLogger(__name__)
+
+
+class KnativeOperatorCharm(CharmBase):
+    """A Juju Charm for Training Operator"""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+        self.interfaces = ""
+
+        self._name = self.model.app.name
+        self._namespace = self.model.name
+        self._operator_service = "/ko-app/operator"
+        self.env = Environment(loader=FileSystemLoader('src'))
+        self._container = self.unit.get_container(self._name)
+        self._resources_files = {
+             "auth_manifests.yaml",
+             "service_account.yaml",
+             "crds_manifests.yaml",
+             # Skipping config manifests for now
```

Comment:
Should we add a manifest that creates the `istio-system` namespace to address the V1.1.x bug?

Reply:
+1